packages/core/src/__tests__/useOnAction.test.tsx

```diff
@@ -12,8 +13,19 @@ import MockRouter, {
   MockActions,
   MockRouterKey,
 } from './__fixtures__/MockRouter';
+import type { NavigationContainerRef } from '../types';
 
-beforeEach(() => (MockRouterKey.current = 0));
+jest.mock('nanoid/non-secure', () => {
+  const m = { nanoid: () => String(++m.__key), __key: 0 };
```

Comment:
```suggestion
  return { nanoid: () => String(++m.__key), __key: 0 };
```

Reply:
Return won't work. The variable is used inside the object.